src/share/classes/com/sun/javatest/regtest/agent/MainWrapper.java

```diff
@@ -70,6 +72,18 @@ public static void main(String[] args) {
 
         // RUN JAVA IN ANOTHER THREADGROUP
         MainThreadGroup tg = new MainThreadGroup();
+
+        Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
```

Comment:
I think if we do decide to set a JVM level default uncaught exception handler then it has to be done only for the case where the `Thread` instance returned by the test thread factory (if there is one) is a virtual thread. So something like:

```java
if (t.isVirtual()) {
  Thread.setDefaultUncaughtExceptionHandler(.....)
}
```
That way we don't interfere with uncaught exception handlers when platform thread is being used.

Reply:
Finally, the major issue with setting a JVM level uncaught exception handler is that, tests will no longer be able to expect the `Thread.getDefaultUncaughtExceptionHandler()` to ever be `null`, even if they launch in `/othervm` mode. I think that's a problem. I can't recollect if jtreg testing framework has other similar cases where it interferes with the tests even with the usage of `/othervm`.

Reply:
No, the plan is to catch all unhandled exceptions. The virtual threads are only a particular case but might be the most common. However, I think we shouldn't silently ignore any of the unhandled exceptions. Some exceptions are already handled using ThreadGroup, and the fix doesn't break this, just add more handling.

Reply:
> Finally, the major issue with setting a JVM level uncaught exception handler is that, tests will no longer be able to expect the Thread.getDefaultUncaughtExceptionHandler() to ever be null, even if they launch in /othervm mode. I think that's a problem. I can't recollect if jtreg testing framework has other similar cases where it interferes with the tests even with the usage of /othervm.

I don't think it is a major issue.  It might be useful to document this, so tests don't expect global UHE to be null.  They could re-write UHE if they need and don't need to preserve/restore the original one.

Reply:
> However, I think we shouldn't silently ignore any of the unhandled exceptions. Some exceptions are already handled using ThreadGroup

Do you mean there are uncaught exceptions that aren't currently being handled/reported by jtreg when using platform thread?

Reply:
I hope that we don't have. Most of test platform threads should inherit MainGroup. However, I don't know if it is true for all pools and executors implementations. And there are JDK internal threads (Managment, JVMCi, Debugger, UI, JFR and other). I hope that they handle exceptions, but haven't checked really all of them. The goal of this fix to ensure that we don't miss any of them if they were thrown during test execution.